### ISSUE TITLE
Enhancement: Azure File Service storage driver

### ISSUE THREAD

**Issue by jgreat** [rancher/rancher#7490]
We use Azure File Service for persistent container volume storage.  I submitted a PR adding a rancher-azurefile driver to the rancher/storage project. 

rancher/storage#30

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.3.2` Cattle: `v0.175.8` UI: `v1.2.40` |
|Access|`azuread` `admin`|
|Orchestration|`Cattle`|
|Route|`stacks.index`|

**Comment by wolli-lenzen** [comment]
any plans about this ?

**Comment by qtzar** [comment]
Would love to see this implemented also.

**Comment by adrianmurphycapita** [comment]
+1

**Comment by samattridge** [comment]
I'd be very interested in this too.

**Comment by danielmkraus** [comment]
+1

**Comment by jgreat** [comment]
After 8 months of use I have some pretty serious reservations for anyone wanting to use Azurefile CIFS storage on linux. 

First is that it's just stupid slow. I've tried various tuning and mount options and get <1MB/s transfer rates. Usually it's more like 300KB/s. I've tried on multiple storage accounts and in different regions. If anyone has a combination of mount/kernel options that works better, I'd love to hear it.

Very rarely (once every 2 months or so) I have an issue where the kernel driver holds onto a file lock while all the userland utilities think the mount is gone.  The only solution is to reboot the vm that last had the mount.
